File: JavaFundamentals/Vacation.java
import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int countPeople = Integer.parseInt(scan.nextLine());
        String type = scan.nextLine();
        String day = scan.nextLine();

        double price = 0;

        switch (type) {
            case "Students":
                if (day.equals("Friday")) {
                    price = countPeople * 8.45;
                } else if (day.equals("Saturday")) {
                    price = countPeople * 9.80;
                } else if (day.equals("Sunday")) {
                    price = countPeople * 10.46;
                }
                if (countPeople>= 30) {
                    price = price - price * 0.15;
                }
                break;
            case "Business":
                if (countPeople>= 100) {
                    countPeople -= 10;
                    price = countPeople * price;
                }
                if (day.equals("Friday")) {
                    price = countPeople * 10.90;
                } else if (day.equals("Saturday")) {
                    price = countPeople * 15.60;
                } else if (day.equals("Sunday")) {
                    price = countPeople * 16;
                }
                break;
            case "Regular":
                if (day.equals("Friday")) {
                    price = countPeople * 15;
                } else if (day.equals("Saturday")) {
                    price = countPeople * 20;
                } else if (day.equals("Sunday")) {
                    price = countPeople * 22.50;
                }
                if (countPeople>= 10 && countPeople <=20) {
                    price = price - price * 0.05;
                }
                break;
        }

        System.out.printf("Total price: %.2f", price);

    }
}

File: JS-FrontEnd/JavaScript Basic Syntax/vacation.js
function priceCalculator(countPeople, typeOfGroup, dayOfWeek) {
    
    let pricePerPerson = 0;
    let type;

    if (dayOfWeek === 'Friday') {
        switch (typeOfGroup) {
            case 'Students':
                pricePerPerson = 8.45;
                type = 'Students';
            break;
            case 'Business':
                pricePerPerson = 10.90;
                type = 'Business';
            break;
            case 'Regular':
                pricePerPerson = 15;
                type = 'Regular';
            break;
        }
    } else if (dayOfWeek === 'Saturday') {
        switch (typeOfGroup) {
            case 'Students':
                pricePerPerson = 9.80;
                type = 'Students';
            break;
            case 'Business':
                pricePerPerson = 15.60;
                type = 'Business';
            break;
            case 'Regular':
                 pricePerPerson = 20;
                 type = 'Regular';
            break;
        }
    } else if (dayOfWeek === 'Sunday') {
        switch (typeOfGroup) {
            case 'Students':
                pricePerPerson = 10.46;
                type = 'Students';
            break;
            case 'Business':
                pricePerPerson = 16;
                type = 'Business';
            break;
            case 'Regular':
                pricePerPerson = 22.50;
                type = 'Regular';
            break;
        }
    }

    let priceWithoutDiscount = pricePerPerson * countPeople;
    let finalPrice = 0;

    // DISCOUNTS
    if (type === 'Students') {
        if (countPeople >= 30) {
            finalPrice = priceWithoutDiscount * 0.85;
        } else {
            console.log(`Total price: ${priceWithoutDiscount.toFixed(2)}`);
            return;
        }
    } else if (type === 'Business') {
        if (countPeople >= 100) {
            finalPrice = priceWithoutDiscount - 10 * pricePerPerson;
        } else {
            console.log(`Total price: ${priceWithoutDiscount.toFixed(2)}`);
            return;
        }
    } else if (type === 'Regular') {
        if ( countPeople >= 10 && countPeople <= 20) {
            finalPrice = priceWithoutDiscount * 0.95
        } else {
            console.log(`Total price: ${priceWithoutDiscount.toFixed(2)}`);
            return;
        }
        
    }

    console.log(`Total price: ${finalPrice.toFixed(2)}`);

}


priceCalculator(30, "Students", "Sunday");
priceCalculator(40, "Regular", "Saturday");